### ISSUE TITLE
Script to automatically count character widths

### ISSUE THREAD

**Issue by randomdude999** [RPGHacker/SMW-Workspace#9]
The README mentions that the only way to get the font width tables is by counting the pixels in YY-CHR. I could write a script that does this automatically.

**Comment by RPGHacker** [comment]
Sounds good, as long as it's optional (optional so that we can hand-adjust stuff in the table if needed). If you do, would be nice to also add a parameter for fixed-width fonts which simply enables a table to a single width for each character.

**Comment by randomdude999** [comment]
I was thinking it would just generate a table that you could use in place of the vwffont.asm file. A fixed width font would just be
```
fillbyte !width : fill 256
```
though, right?

**Comment by RPGHacker** [comment]
Yep, generating a table file was also what I was thinking off.
And yeah, that fillbyte should do the trick, although I would still probably let it generate a full width table to make it easy to adjust, so basically something similar to this

```
; Characters 00 to 0F
db !width,!width,!width,!width,!width,!width,!width,!width,!width,!width,!width,!width,!width,!width,!width,!width
; Characters 10 to 1F
db [...]
```

**Comment by randomdude999** [comment]
I got it to work. I didn't add fixed width table generation yet, I'll work on that tomorrow.
[generate_widths.py](https://github.com/RPGHacker/SMW-Workspace/files/1952873/generate_widths.txt) (had to upload as txt because github)
[Here's](https://github.com/RPGHacker/SMW-Workspace/files/1952883/vwffont1.txt) what it outputs for the Secret of Evermore font. There are a few differences, most notably that the script outputs 0 as the space width, which is incorrect. But that's a simple fix.

**Comment by RPGHacker** [comment]
Looks good! And doesn't the patch use a special character to handle spaces, anyways? I remember it doing so.

**Comment by ExE-Boss** [comment]
Space is `$FE`, and the width is handled by the message header.

**Comment by randomdude999** [comment]
[generate_widths.py](https://github.com/RPGHacker/SMW-Workspace/files/1956649/generate_widths.txt)
Fixed width table support! It uses the width of the widest character as the width for all characters.
[vwffont1.asm](https://github.com/RPGHacker/SMW-Workspace/files/1956651/vwffont1.txt)
I chose `!wd` as the define name because then it's 3 characters, just like a 1-byte hex literal, so if you manually override some widths the table will still be aligned.
Should I document this in the manual and make a pull request?
